### ISSUE TITLE
Laser - Update new posting script and HR

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1414]
In GitLab by @kdjstudios on Mar 21, 2019, 08:30

**Submitted by:** Terri

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2019-03-08-27838/conversation

**Server:** External

**Client/Site:** Laser

**Account:** NA

**Issue:** 

If I am reading correctly it does look like the CSV export option is having issues on the Amtelco side; which was causing the formatting issues. I would think if that can be fixed there will be no need to look at any other options.

 

The text file export “AcctStats.TXT” appears to be nicely formatted with ‘tab’ delimiters. So I was able to open it in excel and then save it out as CSV (Attached “AcctStats – Imported to Excell to CSV.csv”). It appears everything is nicely formatted and does not look like it is missing metric values. So I would believe the support team would be able to import from the text file. The only thing that is missing are the headers. We must have headers in the file otherwise the support team cannot create an import, as it does not know which ‘column’ is which. One solution would be to simply manually copy paste the headers in on the first row before uploading to SA Billing; which It looks like you did in the second tab of the “Acct Stats from txt for SA Bill Research.xlsx” file (more on this below). Otherwise, I would check to see if possibly including the headers is something the Amtelco switch can do from the start to eliminate needing manual adjustments.

 

I have a few concerns with the file “Acct Stats from txt for SA Bill Research.xlsx”. The first tab appears to have the headers for the Acct Stats.TXT, but there are two few headers and they do not appear to match the columns in the AcctStats.txt file either from my manual copy paste. If you notice in my attached file “AcctStats – Imported to Excell to CSV With Headers.csv” that we are missing the last two columns FD and FE.

 

As mentioned by Sharon, from the transpose in the second tab in “Acct Stats from txt for SA Bill Research.xlsx”  that Sharon performed it appears that she was somehow able to match up headers to have the same amount of columns; but we both seem to be a bit curious if they actual match. For me, the columns D, E, and F (Company Name, Start Date, End Date) do not appear to contain values that I would expect them to contain. Example company name only has numeric values it in and they do not appear to be ID numbers or a name either.    

 

In summary, I think the simplest solution for all parties would be to have the CSV export option fixed on the Amtelco side. This way no one at Laser is needing to manual adjust a text file, insert headers, etc. and would eliminate any risk for error there. The second option is to insure we have the correct headers for the text file output and the proper way to transpose them onto the text file export if it is not something the Amtelco switch can do.

Full email thread: [original_message__2_.html](/uploads/0d5e0d49e0dda8b02473137264583073/original_message__2_.html)

[AcctStats+-+Imported+to+Excell+to+CSV.csv](/uploads/fa008f089d93cbee4cd65e92ebe55b2e/AcctStats+-+Imported+to+Excell+to+CSV.csv)

[AcctStats+-+Imported+to+Excell+to+CSV+With+Headers.csv](/uploads/275742a15a469817125dd5c46aa5094e/AcctStats+-+Imported+to+Excell+to+CSV+With+Headers.csv)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 21, 2019, 08:31

Sharon wrote:

> I am leaving now for the day and I am not sure if I am working any days next week, it is spring break for us.
> 
> 
>  
> 
> Hopefully I can get back with Amtelco the week of March 18th and get back to you soon after that.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 21, 2019, 08:31

Gary wrote:

> I would like this matter on the agenda for next week

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 21, 2019, 08:33

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 26, 2019, 09:43

This was reviewed today. Gary will further review with Kess.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 21, 2019, 16:46

Sharon wrote:

> I have contacted Amtelco support and it seems that the solution is to export it into an Access database report and then export that into an excel file. I have attached an example I have run on just a few of our accounts.
> 
> The one 'unusual' aspect of this report is that it reports time in milliseconds. Therefore, you must divide the number by 60,000 in order to get the time in minutes. Although we bill most of our customers on number of calls, we still have a handful that we bill on time, including one of our biggest customers.
> 
> Please let me know if you need anything else from me.

[acct+stats+from+Access+to+Excel++for+sa+bill+research.xlsx](/uploads/3f44439cef88d658ed3005cecac84800/acct+stats+from+Access+to+Excel++for+sa+bill+research.xlsx)

I wrote:

> Thank you for the updated report export. Just from a quick look over I am not seeing any issues and it appears to have everything needed for an import. In regards to the milliseconds, we should be able to address that during the importing process and convert it accordingly. We have provided this to the support team for further review and will follow up.
> 
> You mention there are still two steps your performing to get the report export: First export to Access Database and then export to Excel. It sounds to be simpler then the current manual steps your performing, but we just want to make sure that process is still faster/easier then the current process to ensure we are truly improving your billing process

Marking this for discussion. The attached report looks like the best file we have received so far to use for importing. We will want to prioritize and possible create and estimate and timeline for the re-onboarding if this file should suffice.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 30, 2019, 10:21

I will want to reach out to Laser and confirm they want to keep the existing billing codes or if they would like to add new and additional ones. Currently we pull in 15 columns in formulas vs the roughly 200 columns in formulas that the new import contains. Would you expect to keep the same billing codes or want to increase them..


Gary review on priority level.

[Laser_Answering_Service_-_Human_Readable_Mapping.xlsx](/uploads/d8f48058e87589cff4b4a6abc8bf6be1/Laser_Answering_Service_-_Human_Readable_Mapping.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 30, 2019, 12:32

I wrote:

> The support team has started to go over the attached report and are still reviewing it to provide an estimate and timeline. They did present a few questions that we will need your teams input in order to better estimate the timeline.
> 
> 1. Currently it would appear there are roughly 15 billing codes setup on the site edit page; these are currently being imported and calculated accordingly. I have attached the current mapping we have in place for these codes. We would like to confirm if you are wanting to keep those same billing codes and map/match them to the new document; or if you are wanting to create new billing codes? 
> 2. The file attached contains about 150 columns/metrics that can be imported accordingly. The more columns that are needed to be imported the longer the processing time will be on the upload; this we would expect to still be within a few minutes. But if there are only certain columns/metrics that need to be imported this can greatly improve the upload time. If you are aware of certain columns/metrics that are not needed please let us know.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 4, 2019, 09:38

I will want to work with Kess, Gary, and Kravitz once we have an estimate on timeline and cost to present to the client and ensure we have a proper on boarding.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 5, 2019, 04:19

@kdjstudios 

Do aavaz team have to present an estimate now?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 5, 2019, 09:24

@pchaudhary @tim.traylor 

We will still need the Aavaz team to create an estimate on time and timeline, that Dave will then be able to use as reference when writing the proposal to the client. 

Is your team able to create a rough estimate with the information provided, or what else is still needed? I know we are waiting on confirmation for the columns import, but am uncertain how much of an effect that actually has on the timeline.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 20, 2019, 12:03

@amishra @smasih @pchaudhary @tim.traylor 

May we please get an update. I know we have waited a while for this information to come back from Laser, but it is now waiting on us.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 1, 2019, 16:13

@smasih @amishra @pchaudhary @tim.traylor 

Marking this for discussion. We should have an estimate by now this was marked Important.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 2, 2019, 14:49

I followed up with Sharon again and we received the following:

> “Currently it would appear there are roughly 15 billing codes setup on the site edit page; these are currently being imported and calculated accordingly. I have attached the current mapping we have in place for these codes. We would like to confirm if you are wanting to keep those same billing codes and map/match them to the new document; or if you are wanting to create new billing codes?”
>
> * I am not sure how to answer this question. Why would we want to change the codes that are in place?  What do you mean “new document”?
>
> “The file attached contains about 150 columns/metrics that can be imported accordingly. The more columns that are needed to be imported the longer the processing time will be on the upload; this we would expect to still be within a few minutes. But if there are only certain columns/metrics that need to be imported this can greatly improve the upload time. If you are aware of certain columns/metrics that are not needed please let us know.” 
> * The attached document is a copy of the end result spreadsheet that we upload into SA billing each billing cycle. The total calls, patches, total minutes and patch time are the only ones that I believe are being used to calculate a customer’s invoice. However, the other columns are needed to get the total calls and total minute values. I manually add a formula in those two columns.
> * The original implementation of SA Billing with Laser was done with a former employee, and no one who is here now has any familiarity or knowledge regarding how and why it was set up the way it was.
>  
> 
> The reason any of this was brought to your attention is that we were looking for a way to not have to manipulate the raw data into what is in the attached spreadsheet. So, yes, of course, there are many columns in the raw data that we do not use.

> [2019-07-01+monthly+billing.csv](/uploads/90fead2a6b3882245f8a6b5c39a525c5/2019-07-01+monthly+billing.csv)

I wrote:

> Thank you for the follow up. Let me try to add some clarifications below. 
> 
> 1. The new acct stats report you were able to provide us is planned to become the new file imported and replace the one you just attached. In order for us to import this new file we need to know which columns in the new file will go to which Billing Codes in SA Billing. This is due to the fact that the columns in your current billing upload file and the new acct stats file are different. A simple spreadsheet showing which column headers equal which SA Billing Codes should be perfect for the support team to use. This is what we shall refer to as the "Mapping" document (I originally said new document, which was vague).  Also if there are Billing Codes in SA Billing that are needing to be formulated (added/subtracted/etc) together from multiple columns in the new acct stats report we will need to know which of those columns they are and the formula used for the calculation. We will need your team to provide these, as we do not know which columns on your switches new acct stats report align with which columns on the existing upload file. I will work with our support team to generate example mock up that your team may use as a base for the Mapping document. I will follow up with that in a few days. 
> 
> 2. In regards to which columns to upload lets play it safe and initially upload everything. That way every single metric is being uploaded and will be available for use in SA Billing. It will also make this initial process much easier for your team. If we need to later on we can always remove certain unused columns from being uploaded to help improve the upload time.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 2, 2019, 14:52

@smasih @amishra @pchaudhary @tim.traylor 

I have the following drafted mock-up for a mapping document for their calculated billing codes. As Sharon stated above they want to keep their same billing codes. 

[Laser_Billing_Code_Mapping.xlsx](/uploads/9dadc3ebe28e5784a6511b98cd98b40b/Laser_Billing_Code_Mapping.xlsx)

In regards to the Human Readable report above will suffice to the calculated code, while we are planning on uploading all columns on the report, will we need to have the 7000 codes part?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 11, 2019, 10:31

I will update the mapping document and update this and send out to Sharon.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 11, 2019, 12:34

I wrote:

> We have taken your acct stats export report and assigned a 7000 code to each column/header. You can see each of these listed in the attached file named "Laser Export 7000 codes.xlsx". We have also attached the mapping document named "Laser_Billing_Code_Mapping.xlsx" that we will need your team to fill in with the correct values/formulas. The support team will then be able to create an importing process for the new document to calculate your existing SA Billing Codes. The mapping document we started to fill in with what we would presume to be the calculations and codes, but you will need to review and complete this for all the codes. We have below described in more detail each column of the mapping document to help clarify any concerns. 
> 
> Import SAB Billing Code Name - These are the names of your SA Billing Activities.
> SAB  Billing Code Number - These are the codes assigned to your SA Billing Activity.
> Raw Acct Stats Column Header Name(s) - This will be the column header name from the Laser Export 7000 attachment.
> Usage Import Calculation/Formula  - This is the formula used to calculate the SA Billing Activity. It will generally contain one or more of your Column Headers from the Laser Export 7000 attachment.
> 7000 code calculation - This is the same formula as the Usage Import Calculation/Formula, but instead of using the column headers it is using the assigned 7000 code from the Laser export 7000 attachment. 
> 
> Once we have received the completed mapping document back from you, our support team will be able to estimate a timeline for the updated import process. Please let us know if you have any questions or concerns. 

[Laser_Export_7000_codes.xlsx](/uploads/e600f9999935b7c8bfec5e92e666930a/Laser_Export_7000_codes.xlsx)

[Laser_Billing_Code_Mapping.xlsx](/uploads/5e159929e47aefeee152a0fb64b901ed/Laser_Billing_Code_Mapping.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 15, 2019, 15:55

Sharon wrote:

> I have adjusted/corrected the mapping in the attached spreadsheet for the items Laser uses for billing. As you will see some of the codes in our SA Billing database are not currently used.


[Laser_Billing_Code_Mapping+for+SA+Billing.xlsx](/uploads/9f3a5444c33201461f942ffe9929bb46/Laser_Billing_Code_Mapping+for+SA+Billing.xlsx)

@smasih @amishra @pchaudhary 

Is anything else needed for an estimate?

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jul 30, 2019, 06:45

@smasih @pchaudhary @kdjstudios This should take 16-20 hours to complete the posting script and the human readable.  I am approving this amount of time to complete the tasks.  Please have this completed ASAP so Kyle can address testing with the client when he returns next week.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 31, 2019, 05:34

@kdjstudios @tim.traylor 

I have read all over the conversation from the very beginning & found that they still want to import the file with the 150 columns but only need to apply the calculation for the listed 15 columns only. Is it right?

So, we're developing a posting script which will handle the 15 columns only.

* We also would like to know that for the VCC posting script we have handled the prefix & postfix `0` in the resource ID scenario, but as we can see that there is no resource ID column for the Laser posting script, which means we have to find the account by the Account Number (*1^st column of Usage*), so we want to know that should we add that `0` prefix/postfix substitute method for the account number also here?

* We're going to create the Posting script by taking the below file as the Usage sample. Please have a look & inform ASAP, if you need to change something.  (**excluding extra columns**)  
  [2019-07-01+monthly+billing.csv](/uploads/de458098f91afeb93af72d7a53a4d001/2019-07-01+monthly+billing.csv)

* Also on the mapping file provided by [kyle] (@kdjstudios ) on his last comment is so confusing as on that file the Account Number column does also having the 7000 code which should not be the case according to my understanding. Please correct me if I'm wrong here.

* I would also like to inform that we can't develop the HR functionality for this script until we have the proper mapping document.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jul 31, 2019, 06:14

@pchaudhary 
I have read all over the conversation from the very beginning & found that they still want to import the file with the 150 columns but only need to apply the calculation for the listed 15 columns only. Is it right?  **CORRECT**

We also would like to know that for the VCC posting script we have handled the prefix & postfix `0` in the resource ID scenario, but as we can see that there is no resource ID column for the Laser posting script, which means we have to find the account by the Account Number (_1^st column of Usage_), so we want to know that should we add that `0` prefix/postfix substitute method for the account number also here?  **No.  As we noted from checking the current script, they do not have a leading zero check process.**



We're going to create the Posting script by taking the below file as the Usage sample. Please have a look & inform ASAP, if you need to change something.  (**excluding extra columns**)  **This is the correct sample file:
[acct+stats+from+Access+to+Excel++for+sa+bill+research.xlsx](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/3f44439cef88d658ed3005cecac84800/acct+stats+from+Access+to+Excel++for+sa+bill+research.xlsx)**

Also on the mapping file provided by @kdjstudios on his last comment is so confusing as on that file the Account Number column does also having the 7000 code which should not be the case according to my understanding. Please correct me if I'm wrong here.  **You are correct.  We do not display the account number or name with their 7000-series code in the human readable.  Pls look at the current HR to guide you.  While we're pulling in new fields and have new calculations, everything else will be the same.**


Please note that we need this to be a new layout for Infinity and we will find out the version number to assign to this.  But we don't have this yet.  Piyush mentioned that he thinks the ability to display different switch formats for the same site was developed but not released.  We will need to address that before this new format goes live.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 31, 2019, 07:21

@tim.traylor 

Thanks for the update.

We also looked on http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1417#note_40669, where Sharon is mentioning that the report time is in milliseconds.

Should we also have to address that part in our posting script?

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 31, 2019, 07:32

@tim.traylor 

We also would like to inform that we are unable to find the column name in the Usage for the below:

1.  `Check_In_Calls` - activity code: `2003`
1.  `Non_Live_Calls` - activity code: `2007`
1.  `patches`      - activity code: `2010`
1.  `sms`   - activity code - `2011`


* We are confused in some of the above codes as there are many of the similar name columns in the Usage file.

* Please have a look on the Usage column name & Activity name mapping below to confirm:

````ruby
    sec_talk = row["SecTalkTime"]
    sec_hold = row["SecHoldTime"]
    sec_disc = row["SecDiscTime"]
    chk_in_talk = row["CheckTalkTime"]
    chk_in_hold = row["CheckHoldTime"]
    chk_in_disc = row["CheckDiscTime"]
    non_live_talk = row["NonLiveTalkTime"]
    non_live_disc = row["NonLiveDiscTime"]
````

Please confirm

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jul 31, 2019, 07:48

@pchaudhary Piyush, the column headers for 2013 and 2014 are listed right in the document.  How can you not be able to find that?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 31, 2019, 07:51

@tim.traylor 

Sorry, it's my bad to input those also.

But remaining is still I am not able to judge.

Please help.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jul 31, 2019, 08:05

@pchaudhary 

Patches are specifically defined as 7008.
Check in calls: 7006
Non-live calls: 7010

I don't see SMS and will reach out to the client.  Thx

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 31, 2019, 23:31

@tim.traylor 

Thanks for the Update & we would like to remind that we're still waiting on the confirmation of http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1417#note_44763 regarding milliseconds.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Aug 1, 2019, 06:53

@tim.traylor 

* We have developed the Posting Script initially, but there is still one field `sms` is pending to finish as we're waiting from the client to response on that.

* For now (Development Purpose) - We are mentioning the switch as the `amtelco` for the HR purpose.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Aug 3, 2019, 03:17

assigned to @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Aug 3, 2019, 03:20

changed time estimate to 2d 4h

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Aug 5, 2019, 04:36

@tim.traylor 

We've also changed the milliseconds to minutes as suggested by Sharon on his http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1417#note_40669 

* Now, we're only waiting to know the `SMS` code to complete this issue.


* We also would like to confirm that there are 2 activities in the Laser site,
  * Check In Calls - Check_In_Calls
  * Non Live Calls - Non_Live_Calls

which do have the Capital alphabets in its cdr field. So, we just want to confirm that the User/Ops is not supposed to change this in future also.  

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Aug 6, 2019, 02:18

added 6h of time spent

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Aug 6, 2019, 05:31

@tim.traylor 

As discusses this in today's huddle, Tim's suggested us to modify the posting script such that it will act case-insensitive.  
We looked into this & below are the details.  
* This change is possible.  
* But the previous SAB convention was written in such a way that it will match to exactly what we have in the Cdr Field column on the activity of the Site.  
* So, according to my best understanding, this change will affect all of the SAB's while creating the transaction on the draft invoice creation step.  
* So, this is a big change to be done.  
* We need the confirmation from the Tim to proceed in this further as we have to check all of the posting scripts after the change.

**Note** - This is not the issue with the current code, it will happen only if the User changes the Cdr Field column from the Site edit page. This type of scenarios can b handled easily if we made the CDR field column **readonly**.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 7, 2019, 11:04

Sharon wrote:

> I need to clarify that my work schedule is not consistent. I was not here yesterday and today is billing so I will not be able to get to this until next week as I will not be here tomorrow.
> 
> Thank you for your patience

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Aug 8, 2019, 09:33

@kdjstudios pls test this and determine if we can, at any point in invoice processing, a chance for Ops to manually enter activities on the invoice.  We want to know if they typically could create their own activity codes and include on an invoice even though they don't exist in the usage file.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 8, 2019, 09:41

@tim.traylor @pchaudhary 

I was able to test this and the draft invoice is pulling in non-usage file activities and showing with a zero amount. With that in mind I do not believe we should be making the field read only. As I can see operations changing those values over time and they would not have any effect on the import process to my knowledge. 

![image](/uploads/cafbedb16ef100c582c34482d4fc23e1/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Aug 13, 2019, 05:01

@kdjstudios 

I would like to confirm Kyle's statement but at the same time, I also would like to confirm that in this case if the User changes the Quantity from the Draft invoice page, then there will be no effect on the HR report.

* The HR report only effects if the CDR records changes but in this case, there will be no changes in the CDR records as well as in the HR report.
* So, our main concern still having the point that we can disable the CDR fields on the Site edit page.
* But this also attracts our mind towards the point that should we allow to do the quantity changes for the activity on the invoice edit while there is no CDR records generation & which causes to no history maintenance for the long run.
* We can't even guess in future that from where does it come from?

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Aug 14, 2019, 24:50

added 1h 30m of time spent

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 27, 2019, 14:09

changed title from **Laser - {-Raw Acct Stats Report-}** to **Laser - {+Update new posting script and HR+}**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 27, 2019, 14:19

Dave wrote: (Sharons additions are in bold)

> As we are trying to complete your SA Billing request for the New Posting Script please reply and confirm the below ASAP.
> 
> Please take a few minutes to review the 7000 codes we will be using for your data. I know that several of the formulas had already been filled out, but many of the individual fields were not identified in the new file layout. **We do not use the other fields, I have no idea why they are in our SA Billing. That happened years before I started with Laser and no one is still here from that time who was involved in setting up SA Billing.**
> 
> Also, we noticed that in the minute’s calculation for your 2013 code you have 6 fields specified for time, but only 3 fields for count (the 2014 code). Can you please confirm that is accurate? **Yes this is correct**
> 
> Lastly, we don't see any field in the new file that is clearly SMS (code 2011). Can you please let us know which column that is? **There is no column for this**
> 
> We are pushing the development team to have this ready so Kyle can start the confirmation testing with you when he gets back from vacation next week, so if you have a minute to get this back to us ASAP that would be awesome.
> 
> Also Sharon,
> Would you also be able to inform us which version of Amtelco you are using and if Amtelco provided any documentation for this report please forward that to us. We are hoping to be able to better identify what each column/metric is and help improve our support for the Amtelco switch. 
> Thank you and we greatly appreciate your time and aid.  


Sharon's updates:

> Before I spend a lot of time on the spreadsheet, I want to clarify a few things.
> The reason I did not fill out the rest of the spreadsheet is that those items are not relevant. By that I mean the only ‘SAB Billing Code Number’ that we utilize for billing are:
> 
> Patches 2010
> Minutes Allowed 2013
> Calls Allowed 2014
> Patch Time 2020
> 
> Some of the others are used in the formulas as I entered already. Of course, the Resource ID is used by SAB for import purposes. 
> 
> Other items like SMS are manually input as a recurring fee and are NOT something we get from the Acct Stats report, therefore there is no column for that.
> 
> Yes, minutes calculation has 6 fields in the formula and call count is only 3 fields used for calculation.
> 
> If you need further clarification, please call me before 1:00pm CST so we can talk about it over the phone. 979-361-3303


> Sorry forgot to include the version of Amtelco 5.1.6508.49

Terri wrote:

> We need to get this clarified asap Kyle.  Can you get on a call with her to review?  She is leaving her office at 2p EST.  
> 
> If she only cares about those four fields great, but we want to be sure!


I wrote:

> I apologize for the delay in response, as I just now saw this. I have been on a meeting with John all day reviewing SAB tickets and did not see this. I do think Tim, John, and I should review these responses before we reach back out to Sharon. I have this marked for discussion during tomorrow’s SA Billing Meeting. 
> 
> In a quick review of Sharon’s responses it does look like we have everything we need in order to proceed. 
> 
> NOTE: I added the HD to this so we can keep this email thread documented on the ticket: 2019-03-08-27838

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 5, 2019, 15:46

@tim.traylor 

I forgot to bring this up on today's meeting. I just want to confirm we are needing anything more from Laser?

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Sep 6, 2019, 05:38

@kdjstudios @jredfield Hi Kyle, this is really your baby - I just stepped in to escalate when we were not getting what we needed.  With the experience you have you should definitely be able to determine if we need anything else and if Aavaz has everything they need.  If you get stuck on anything please hit me up.  Thx

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 6, 2019, 08:24

@tim.traylor @jredfield 

Thanks, I didn't see any thing else needed, so I initially put this back into an estimate from the Aavaz team. 

Side question: It looks like that while I was on vacation this was marked directly for development without an estimate. http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1417#note_44785 Does this mean we do not need an estimate to provide to Dave and Laser? Also if so, I presume if no estimate was needed, we are not billing the client for this then either?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 12, 2019, 02:07

@kdjstudios 

We're consolidating the ticket here.

* The initial request is to create a new posting script. We have almost completed the development as per the http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1417#note_44785. The only remaining thing is one of the code `sms`
* Then at a point during the development we found that the CDR code of the activity should not be changed without the consent of the developer as the code also need to be changed in the posting script for its effect.
* We proposed the solution to disable the CDR field column so that the end-user can't be able to change it. http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1417#note_44984
* Then [Tim] (@tim.traylor ) asked Kyle to check that a chance for Ops to manually enter activities on the invoice, to check that if there would be any negative impact on disabling the CDR field column. http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1417#note_45192
* [Kyle] (@kdjstudios ) checks & found that draft invoice is pulling non-usage file activities and showing with a zero amount & operations can change these values over time & it did not have any effect on the import process. http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1417#note_45193
* We confirm Kyle's statement & also stated that if the user changes anything on the invoice then there'll be no change in the CDR records even in the HR report.
* So, by doing the above tests we found that this can cause a big issue in the near future as we don't track the record of change & the HR report will also not show this changed value. http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1417#note_45284
* Our main concern is still on, that we still can disable the CDR record as there'll be no harm because by changing the quantity direct from the draft invoice will not show any effect on the CDR or HR.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 12, 2019, 02:19

@kdjstudios 

We are not sure why this is in development label? As mentioned earlier, we have almost completed the development & waiting for the discussion on the CDR field column thing. 
We can discuss this in our Thursday meeting as this is already having the Thursday label.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 12, 2019, 08:44

@pchaudhary @tim.traylor @jredfield @smasih 

* We are not sure why this is in development label? As mentioned earlier, we have almost completed the development & waiting for the discussion on the CDR field column thing. We can discuss this in our Thursday meeting as this is already having the Thursday label.

> You were the one who moved this to development? How can you not know why?

Also we still need a estimate, as since we never got that there has been no approval for any work to be done one this from the client. (Although I sure hope we do get their approval after what your stating). NOTE: Please do not remove the only "PL" label on tickets it only makes it harder to track and find.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 5, 2019, 13:08

@smasih @amishra @jredfield 

Please post an update on what is still needing to be completed on this. If I recall correctly this was later approved by David and Gary this would not be charged to the client, so we are no longer needing an estimate. From reviewing the ticket, It appears all development was completed with out an estimate and there was only a concern in regards to something dealing with CDR codes. Please explain further, so we may communicate this with the client.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 7, 2019, 02:49

@kdjstudios @jredfield 

Thank you for the update.

I have read all over the conversation and As per Piyush's [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1417#note_47306) and [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1417#note_47305), we have almost completed the development & waiting for the discussion on the CDR field column thing.

Please review & let us know if anything else to do from my end.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 7, 2019, 10:03

@amishra @jredfield 

Thank you for the update. I am going to create a new ticket about the CDR column on the Site edit page. As that not only effect this but all sites/clients. So it should be addressed on it's own #1631 .

Please leave the CDR as is for this and mark when this is ready for testing.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 18, 2019, 06:44

@kdjstudios @jredfield 


Thank you for the update.

 We have setup a branch on the merge server for your testing.

Please test it.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 18, 2019, 11:50

@jredfield 

I believe we will want to have a meeting with Laser and review the updated process and allow for them to do a "side by side" billing cycle to ensure all the metrics are calculated exactly as they should be. However, we may simply want to have the support team setup the dev server with Laser's last billing cycle removed, and we can run that billing cycle to do a first check?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 26, 2019, 13:41

I uploaded the attached usage file (I had to convert to csv from excel: [Laser_new_usage.csv](/uploads/4ccbc22e28ae32b3f7ef4b0a8139e3e6/Laser_new_usage.csv)) but I uploaded it to the Maser billing cycle, however the first account 2101 Distribution Now is part of the Monthly billing cycle, not the Master. It did not however show on the Missing accounts. Should this have shown on the Missing Accounts, since it was not part of that billing cycle? or is the missing accounts to show just accounts that are not in SA Billing at all? - This would be probably for Gary to confirm.


The apply step, took about a minute to do for the Monthly Billing cycle. Is there anyway we can speed this up? It appears there was only roughly 200 invoices/accounts created, which I would believe we should be able to do in less then 30 seconds. [billing_cycle_detail_11_28_2019_Monthly.csv](/uploads/e68124ca49f999918ef1bb81de2e091c/billing_cycle_detail_11_28_2019_Monthly.csv)


The HR report does not appear to be working. Also it does not appear that any of the usage is being put on the account's invoice. It would appear after looking at the download usage file from SAB, that our posting script is not working and is calculating everything as zeros, also all of the raw columns are missing too. Upon further review it also looks like the columns that are there, are duplciated, but have different formating in the headers.  (Here is the downloaded copy of the usage file from SAB: [laser_new_usage.csv.20191126153118.20191126153118.csv](/uploads/15d0a64792a6c6605de3bf7d2c874190/laser_new_usage.csv.20191126153118.20191126153118.csv) )

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 26, 2019, 13:44

@smasih @amishra 

@jredfield and I were able to review and start to test this, but it appears it is not working. Please see above and research and respond.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Nov 26, 2019, 21:51

assigned to @amishra and unassigned @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Nov 26, 2019, 22:23

assigned to @ranand and unassigned @amishra

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Nov 27, 2019, 06:04

@kdjstudios According to @ranand another fix was put in place so this is ready to be tested again.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Nov 27, 2019, 06:41

@kdjstudios @jredfield 

We have fixed the issues. And put it again for the testing on the dev server.

**FYI:**

Codes and calculations for SMS and Secretarial Hold usage are missing.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2019, 11:24

@ranand @jredfield 

Thank you for the update. I was able to test this and all appears to be working exactly as originally spec'd. However, I am seeing that we may have a concern, as I think only the metrics with "Time" in them are to be divided by 60,000 to convert them to minutes, I believe all other metrics are not "time" related.  As you can see in my attachment ([Laser_Upload_Calculation_Checks.xlsx](/uploads/d4f652dce1262734e04cf4355234dd5c/Laser_Upload_Calculation_Checks.xlsx)), the value in the metrics that do not have the "time" label in them appear to be substantially lower, and thus when we do the conversion to minutes they end up being always equaling "1". I believe I can confirm this by looking at which metric they use in the RPM report for a count. Which is the total of those three metrics that do not have the "time" label in the header.

May we please only do the conversion to minutes for the metrics that contain the label "time" in the header? After that we should be good to bring this back to Laser for their testing and feedback.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 28, 2019, 24:21

@kdjstudios @jredfield 

Thank you for your valuable feedback.

We have Updated the Posting Script & HR report Accordingly as you requested above that only conversation to minutes which contains the label "time" in the header. Please test it on the merge server.


**Before Posting Script**
`````ruby
    sec_op = ((row["SecOp"])/60000).ceil
      # => 7004

    dial_out_op = ((row["DialoutOp"])/60000).ceil
      # => 7077

    patch_no_op_time = ((row["PatchNoOpTime"])/60000).ceil
      # => 7015

    patch_op_time = ((row["PatchOpTime"])/60000).ceil
      # => 7016

    sec_talk = ((row["SecTalkTime"])/60000).ceil
      # =>  7027 ===============> (2000)

    sec_hold = ((row["SecHoldTime"])/60000).ceil
      # =>  7030 ===============> (2001)

    sec_disc = ((row["SecDiscTime"])/60000).ceil
      # =>  7033 ===============> (2002)

    check_in_calls = ((row["CheckOp"])/60000).ceil
      # =>  7006 ===============> (2003)

    chk_in_talk = ((row["CheckTalkTime"])/60000).ceil
      # =>  7028 ===============> (2004)

    chk_in_hold = ((row["CheckHoldTime"])/60000).ceil
      # => 7031 ===============> (2005)

    chk_in_disc = ((row["CheckDiscTime"])/60000).ceil
      # =>  7034 ===============> (2006)

    non_live_calls = ((row["NonLiveOp"])/60000).ceil
      # =>  7010 ===============> (2007)

    non_live_talk = ((row["NonLiveTalkTime"])/60000).ceil
      # =>  7029 ===============> (2008)

    non_live_disc = ((row["NonLiveDiscTime"])/60000).ceil
      # =>  7035 ===============> (2009)

    patches = ((row["PatchOp"])/60000).ceil
      # =>  7008 ===============> (2010)

    total_minutes = sec_talk + chk_in_talk + non_live_talk + sec_disc + chk_in_disc + non_live_disc
      # => 7027 + 7028 + 7029 + 7033 + 7034 + 7035 ===============> (2013)


    total_calls = sec_op + check_in_calls + dial_out_op
      # => 7004 + 7006 + 7077 ===============> (2014)


    patch_time = patch_no_op_time + patch_op_time
      # => 7015 + 7016 ===============> (2020)
``````

**Updated Posting Script**

`````ruby
sec_op = (row["SecOp"])
      # => 7004

    dial_out_op = (row["DialoutOp"])
      # => 7077

    patch_no_op_time = ((row["PatchNoOpTime"])/60000).ceil
      # => 7015

    patch_op_time = ((row["PatchOpTime"])/60000).ceil
      # => 7016

    sec_talk = ((row["SecTalkTime"])/60000).ceil
      # =>  7027 ===============> (2000)

    sec_hold = ((row["SecHoldTime"])/60000).ceil
      # =>  7030 ===============> (2001)

    sec_disc = ((row["SecDiscTime"])/60000).ceil
      # =>  7033 ===============> (2002)

    check_in_calls = (row["CheckOp"])
      # =>  7006 ===============> (2003)

    chk_in_talk = ((row["CheckTalkTime"])/60000).ceil
      # =>  7028 ===============> (2004)

    chk_in_hold = ((row["CheckHoldTime"])/60000).ceil
      # => 7031 ===============> (2005)

    chk_in_disc = ((row["CheckDiscTime"])/60000).ceil
      # =>  7034 ===============> (2006)

    non_live_calls = (row["NonLiveOp"])
      # =>  7010 ===============> (2007)

    non_live_talk = ((row["NonLiveTalkTime"])/60000).ceil
      # =>  7029 ===============> (2008)

    non_live_disc = ((row["NonLiveDiscTime"])/60000).ceil
      # =>  7035 ===============> (2009)

    patches = (row["PatchOp"])
      # =>  7008 ===============> (2010)

    total_minutes = sec_talk + chk_in_talk + non_live_talk + sec_disc + chk_in_disc + non_live_disc
      # => 7027 + 7028 + 7029 + 7033 + 7034 + 7035 ===============> (2013)


    total_calls = sec_op + check_in_calls + dial_out_op
      # => 7004 + 7006 + 7077 ===============> (2014)


    patch_time = patch_no_op_time + patch_op_time
      # => 7015 + 7016 ===============> (2020)
`````

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 29, 2019, 10:36

@amishra 

May we please update the HR report to show on the two metrics 2020 and 2013 the division by 60000 only once. Like such: "(PatchNoOpTime + PatchOpTime) / 60000". This should look a bit less confusing. 

![image](/uploads/299a5ab6e9d8ce82f67c5182052c4888/image.png)

@jredfield 

I think next week we should be good to reach out to Laser and setup a time to do some side by side testing.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 29, 2019, 11:48

@kdjstudios @jredfield 

Thank you for the update

I would like to inform you that the current code `2020`as:
````ruby 
code 2020:  `["PatchNoOpTime"]/60000).ceil + ["PatchOpTime"])/60000).ceil`   
  `````
**Do we need to change the calculation?** 

Please look updated Posting Script and let me know if we need to change in the posting script or calculation for the codes.

`````ruby
sec_op = (row["SecOp"])
      # => 7004

    dial_out_op = (row["DialoutOp"])
      # => 7077

    patch_no_op_time = ((row["PatchNoOpTime"])/60000).ceil
      # => 7015

    patch_op_time = ((row["PatchOpTime"])/60000).ceil
      # => 7016

    sec_talk = ((row["SecTalkTime"])/60000).ceil
      # =>  7027 ===============> (2000)

    sec_hold = ((row["SecHoldTime"])/60000).ceil
      # =>  7030 ===============> (2001)

    sec_disc = ((row["SecDiscTime"])/60000).ceil
      # =>  7033 ===============> (2002)

    check_in_calls = (row["CheckOp"])
      # =>  7006 ===============> (2003)

    chk_in_talk = ((row["CheckTalkTime"])/60000).ceil
      # =>  7028 ===============> (2004)

    chk_in_hold = ((row["CheckHoldTime"])/60000).ceil
      # => 7031 ===============> (2005)

    chk_in_disc = ((row["CheckDiscTime"])/60000).ceil
      # =>  7034 ===============> (2006)

    non_live_calls = (row["NonLiveOp"])
      # =>  7010 ===============> (2007)

    non_live_talk = ((row["NonLiveTalkTime"])/60000).ceil
      # =>  7029 ===============> (2008)

    non_live_disc = ((row["NonLiveDiscTime"])/60000).ceil
      # =>  7035 ===============> (2009)

    patches = (row["PatchOp"])
      # =>  7008 ===============> (2010)

    total_minutes = sec_talk + chk_in_talk + non_live_talk + sec_disc + chk_in_disc + non_live_disc
      # => 7027 + 7028 + 7029 + 7033 + 7034 + 7035 ===============> (2013)


    total_calls = sec_op + check_in_calls + dial_out_op
      # => 7004 + 7006 + 7077 ===============> (2014)


    patch_time = patch_no_op_time + patch_op_time
      # => 7015 + 7016 ===============> (2020)
`````

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 29, 2019, 12:29

@amishra 

Yes, as requested please make it add the metrics and then do the division. it should be as below. Please do this for both 2013 and 2020

code 2020: (`["PatchNoOpTime"].ceil` + `["PatchOpTime"]).ceil`) / 60000

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 29, 2019, 12:35

@kdjstudios @jredfield 

Thank you for the confirmation.

We will change this HR report &  the posting script calculation too.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 1, 2019, 23:49

@kdjstudios @jredfield 

We have updated the posting Script & Hr report for code `2020` and `2013`. Please have a look.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 2, 2019, 08:49

@amishra @jredfield 


Thank you, the HR looks substantially more professional. I think we are ready to do a side by side billing and test with Laser. @amishra @smasih may we please have this server updated to be a current copy of the external production (with these updates) this way we can have laser do a side by side billing to ensure that all metrics are being calculated correctly.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Dec 3, 2019, 04:05

@kdjstudios @jredfield 

Do you want us to ER-release it or do you want us to set up a branch on the testing server with the latest data of the hosted server?

If we release this then we need to do followings:
* We will need to change the site switch to "Amtelco" for the Laser site.
* We will need to change the posting Script for the Laser site.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 3, 2019, 12:33

@ranand 

Please do NOT release anything to production.

We need to have the latest data from the hosted server copied over to a testing/dev server to do the side by side billing to confirm everything is working.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 4, 2019, 06:21

@kdjstudios @jredfield 

Thank you for the update.

We have restored the latest data on the merge server. Please test it.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 4, 2019, 10:18

@amishra @ranand @rranjan @smasih 

Thank you, Please do not alter anything on the Merge server. @jredfield and I will be scheduling a call with the client to review the updated process and do a side by side billing. I would expect this could take a week or so, due to Sharon's part time availability.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 4, 2019, 10:26

I wrote:


> We believe the updated import process is ready for your teams initial testing and feedback.  We are reaching out to see if you have an hour of time this week or next week to join a screen-share and review the updated process. As part of this we will do a side by side billing with your December billing cycle on production so we may then compare and confirm everything is working as expected. Keeping in mind that your schedule tends to vary, we would request if you could provide a few dates and times that you're available so we can confirm with our team which one will work best for everyone. Thank you and we look forward to your response. 
> 
> We also are waiting for feedback to the following from your team.
> 
> 1. We have a concern we found while testing that we are still missing the values to be associated with the two following Billing Codes. If you could please provide which header/column metric(s) from the report they should be pulling from we will update the importing process to calculate those also.
>  - SMS  2011
>  - Secretarial Hold 2001
> 
> 2. We are following up to know which Amtelco version you are using and if possible if you have any documentation on the report that you are able to send to us.
> 
> We thank you again for you patience and aid in helping us better SA Billing to accommodate and ease your billing process. Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 6, 2019, 14:27

Sharon wrote:

> There is no column for SMS 2011 nor Secretarial hold 2001. We do not bill for Secretarial hold so that can be removed from SA billing. The SMS 2011 is a recurring fee and does not come from the Amtelco reports.
> 
> We will have to get back with you regarding possible times for us to join in on a screen share.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 6, 2019, 14:28

@ranand 

We should be good with now using the merge server for other testing items. Once we hear back from Sharon on a date and time, then we will just need to push this back to a dev server.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 8, 2019, 22:48

@kdjstudios @jredfield 

Thank you for the confirmation.

Please let us know when we push back for your/client testing.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 12, 2019, 11:23

I wrote:

> We are just following up to see if you could please provide a few dates and times to review the updated progress. 
> 
> NOTE: If we are able to review this and receive your teams approval before Dec 20th , we believe the support team will be able to release this before the end of December, which would mean at the begging of the new year your team would be able to start billing using the updated and quicker process.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 16, 2019, 13:02

This is placed on hold until Sharon returns the first week of Jan.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 20, 2020, 14:59

I wrote:

> We are just following up to see if you have been able to settle back in from all the holiday fun and would be ready to setup a call/demo to review new importing process?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 30, 2020, 11:51

Moving to pause until we hear back from the client.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 3, 2020, 10:44

Removing Priority until we hear from the client.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 15, 2020, 09:17

Marking for Thursday discussion to determine how we should continue to follow up on this.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 24, 2020, 13:50

I wrote:

> We are following up on older requests that we have not had a response on. We are wondering if your team was still interested in the updated uploading process? If so, we should be able to setup a demo/call to review it in the upcoming weeks to make sure it works before we release it. If not, please let us know so we may close this. Thank you and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 25, 2020, 07:51

Walter wrote:

> We are not in a position to work on this now.

I wrote:

> We will be closing this request for now. Please reply back when your team is ready.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 25, 2020, 07:51

closed
